src/lsp/cobol_lsp/lsp_request.ml

```diff
@@ -677,6 +677,35 @@ let handle_codelens registry ({ textDocument; _ }: CodeLensParams.t) =
     end
   |> Option.value ~default:[]
 
+(** { Rename } *)
+
+exception CopybookRenameError
+
+let handle_rename registry (params: RenameParams.t) =
+  let { textDocument; position; newName = newText; _ }: RenameParams.t = params in
+  let doc = Lsp_server.find_document params.textDocument registry in
```

Comment:
There are probably exception cases to be dealt with here. (maybe a use of the usual helpers from the top of this file is advisable; probably `try_with_main_document_data`).

Reply:
I specifically removed this `try` function because I wanted the function to throw but I change the behavior later, I will put back the wrapper